sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/DisposeFromDispose.cs

```diff
@@ -93,7 +92,10 @@ private static bool IsDisposeMethodCalled(InvocationExpressionSyntax invocation,
 
             var disposeMethod = DisposeNotImplementingDispose.GetDisposeMethod(semanticModel.Compilation);
             return disposeMethod != null &&
-                methodSymbol.Equals(methodSymbol.ContainingType.FindImplementationForInterfaceMember(disposeMethod));
+            (
+                methodSymbol.Equals(methodSymbol.ContainingType.FindImplementationForInterfaceMember(disposeMethod)) ||
+                methodSymbol.ContainingType.IsDisposableRefStruct()
```

Comment:
We should check for the method symbol signature here too in case of ref structs, otherwise we will get FP:
```
    public ref struct MyTestDisposableRefStruct
    {
        public void Dispose(){}
        public void AnotherMethod(){}
    }

    public ref struct RefStructResourceHolder
    {
        // ref structs can only be fields in other ref structs
        private MyTestDisposableRefStruct foo;

        public void CleanUp()
        {
            this.foo.AnotherMethod(); // FP
        }

        public void Dispose()
        {}
    }
```

Reply:
you saved the day!